tests/examples/llm/common/test_content_extractor_module.py

```diff
@@ -0,0 +1,143 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+import random
+import shutil
+import string
+import tempfile
+import types
+import uuid
+from functools import partial
+from typing import Callable
+from typing import Dict
+from typing import Generator
+from typing import List
+
+import fsspec.core
+import pandas as pd
+import pytest
+
+from morpheus.config import Config
+from morpheus.messages import MessageMeta
+from morpheus.pipeline import LinearPipeline
+from morpheus.stages.general.linear_modules_stage import LinearModulesStage
+from morpheus.stages.input.in_memory_data_generation_stage import InMemoryDataGenStage
+from morpheus.stages.output.in_memory_sink_stage import InMemorySinkStage
+
+logger = logging.getLogger(f"morpheus.{__name__}")
+
+
+class TempCSVFiles:
+
+    def __init__(self, num_files: int, columns: Dict[str, Callable[[], any]]):
+        self.temp_dir = None
```

Comment:
I would update the test to use the `tmp_path` fixture provided by pytest, and then make `tmp_dir` a constructor argument. pytest will ensure that the `tmp_path` is cleaned up on test completion, removing the need to call rmtree.